### PR TITLE
Search: allow OR in parenthesis

### PR DESCRIPTION
**Pull request by ronso0**
Before we had to do this
`house deep | house minimal | house tech`
which makes BPM/genre OR queries quite long.

Now we can do
`house (deep|minimal|tech)`
`hop (cm:pop|title:hip)`

Quotation detection etc. is also still intact :tada: 
_Theoretically_ it should also work with deeper levels of nesting, but I didn't test this and also don't think it matters in real life, hence we shouldn't add such claims to the docs.

- [x] add tests
- [ ] remove TRACE commit
_____
Disclaimer: I did this with an agent since I wanted this badly but regex and complex parsing is certainly not my cup of tea. First I tried the regex approach, but turned to clean methods for manual splitting (the regex became monstruos when I noticed we also need to cover quotes as well as a few edge cases -- "if a PhD is required to understand the regex, drop it" )

### CODE REVIEW

**Comment by ronso0** [comment]
I added test for
* regular usage of course `a (1|2) | b (8|9)`
* | in quotes
* unclosed parenthesis
* literal `(|)` (who knows what ASCII art folks use to tag their music)

had to switch to raw C strings because of issues with the escaped quotes \"

I wasn't particularly creative with the query examples, so let me know if you have more elaborate (real life) queries to test, or if you think we need to test other special cases.

**Comment by ronso0** [comment]
Okay, so the first real test went good.

**Nice to have:**
prefix parenthesis with shortcuts
`bpm:105-115 c:(bass|tribal|groove)` would search for BPM and OR for comments

File: src/library/searchqueryparser.cpp [hunk 1]
```diff
@@ -290,28 +323,53 @@ void SearchQueryParser::parseTokens(QStringList tokens,
                             m_pTrackCollection->database());
                 }
             }
+        } else if (token.startsWith("(") && hasClosingParenthesis(token, tokens)) {
```
Comment (acolombier) [inline src/library/searchqueryparser.cpp:326]: How would that behave with something like `artist:"(DJ Flop remix)"`, which previously aimed to search for the `(DJ Flop remix)` literal.

Comment (ronso0) [reply src/library/searchqueryparser.cpp:326]: Exactly as before, only tokens starting with `(` are taking the new parenthsesis path.
* with the text filter `artist:` it takes the text filter path.
tested with `t:" (Origxxx)"`: :heavy_check_mark: works 
* without preceeding text filter, ie. `|` in the quoted parenthesis part, it takes the 'untagged' path
Tested with ` (Orig | xxx)`: :heavy_check_mark:  works

I can add a test for that, too.
